src/sonic-yang-models/yang-models/sonic-dtel.yang

```diff
@@ -0,0 +1,294 @@
+module sonic-dtel{
+    yang-version 1.1;
+    namespace "http://github.com/Azure/sonic-dtel";
+    prefix dtel;
+
+    import ietf-inet-types {
+        prefix inet;
+    }
+    import sonic-port {
+        prefix port;
+        revision-date 2019-07-01;
+    }
+    import sonic-portchannel {
+        prefix lag;
+    }
+    import sonic-vrf {
+        prefix vrf;
+    }
+
+    organization
+        "SONiC";
+
+    contact
+        "SONiC";
+
+    description
+        "SONIC DTEL";
+
+    revision 2022-06-16 {
+        description
+            "Initial revision.";
+    }
+
+    container sonic-dtel {
+        container DTEL {
+            description "DTEL part of config_db.json";
+
+            list DTEL_LIST {
+                key "name";
+
+                leaf name {
+                    type string {
+                        length 1..255;
+                    }
+                }
+
+                leaf switch_id {
+                    type uint8;
+                    description "Globally unique switch ID";
+                }
+
+                leaf flow_state_clear_cycle {
+                    type uint16;
+                    description "DTEL flow state clear cycle, units: seconds";
+                }
+
+                leaf latency_sensitivity {
+                    type uint16;
+                    description "Latency sensitivity for flow state change detection, units: 2^n nanoseconds";
+                }
+
+                leaf-list sink_port_list {
+                    type union {
+                        type leafref {
+                            path "/port:sonic-port/port:PORT/port:PORT_LIST/port:name";
+                        }
+                        type leafref {
+                            path "/lag:sonic-portchannel/lag:PORTCHANNEL/lag:PORTCHANNEL_LIST/lag:name";
+                        }
+                    }
+                    default "";
+                    description "DTEL sink ports";
+                }
+
+                leaf int_endpoint {
+                    type string {
+                        pattern "true|false";
+                    }
```

Comment:
please use type boolean

Reply:
Got it. Already fixed it.
Please review again.